Update Spring Data JPA to use spring-dotenv (#1050)
oktadeveloper/okta-blog: _source/_posts/2020-11-20-spring-data-jpa.md

image: blog/spring-data-jpa/secure-spring-data.png
type: conversion
changelog:

> - 2021-01-21: Updated to use `.okta.env` instead of `application.properties`. This change is necessary since the okta-spring-boot-sample now uses [dotenv to load Okta settings](https://github.com/okta-samples/okta-spring-boot-sample/pull/22). Changes to this post can be viewed in [okta-blog#1050](https://github.com/oktadev/okta-blog/pull/1050).

Author: ```suggestion
- 2021-01-25: Updated to use `.okta.env` instead of `application.properties`. This change is necessary since the okta-spring-boot-sample now uses [dotenv to load Okta settings](https://github.com/okta-samples/okta-spring-boot-sample/pull/22). Changes to this post can be viewed in [okta-blog#1050](https://github.com/oktadev/okta-blog/pull/1050).
```

image: blog/spring-data-jpa/secure-spring-data.png
type: conversion
changelog:

> - 2021-01-25: Updated to use `.okta.env` instead of `application.properties`. This change is necessary since the okta-spring-boot-sample now uses [dotenv to load Okta settings](https://github.com/okta-samples/okta-spring-boot-sample/pull/22). Changes to this post can be viewed in [okta-blog#1050](https://github.com/oktadev/okta-blog/pull/1050).

Reviewer: 2022
```suggestion
- 2022-01-25: Updated to use `.okta.env` instead of `application.properties`. This change is necessary since the okta-spring-boot-sample now uses [dotenv to load Okta settings](https://github.com/okta-samples/okta-spring-boot-sample/pull/22). Changes to this post can be viewed in [okta-blog#1050](https://github.com/oktadev/okta-blog/pull/1050).
```

## Generate an Access Token JWT

> You used Okta as your OAuth 2.0 and OpenID Connect (OIDC) provider when you used the Okta CLI. The CLI created an OIDC application for you on the Okta developer site and filled in the config details in a `.okta.env` file. Your application is now expecting a JSON Web Token (JWT) when you make requests.

Reviewer: should we also remind people to gitignore the .env files?

Reviewer 2: ````suggestion
You used Okta as your OAuth 2.0 and OpenID Connect (OIDC) provider when you used the Okta CLI. The CLI created an OIDC application for you on the Okta developer site and populated the config details in a `.okta.env` file. Your application is now expecting a JSON Web Token (JWT) when you make requests.

**IMPORTANT:** Keep your secrets out of source control by adding `okta.env` to your `.gitignore` file: 

```bash
echo .okta.env >> .gitignore
```
````

This probably has too many inline
